pkg/ddl/resource_group.go

```diff
@@ -222,6 +222,8 @@ func SetDirectResourceGroupSettings(groupInfo *model.ResourceGroupInfo, opt *ast
 	case ast.ResourceGroupRunaway:
 		if len(opt.RunawayOptionList) == 0 {
 			resourceGroupSettings.Runaway = nil
+		} else {
+			resourceGroupSettings.Runaway = &model.ResourceGroupRunawaySettings{}
```

Comment:
This part changed the behavior, for the old code, if `resourceGroupSettings.Runaway` is not nil, it won't be overwritten in `SetDirectResourceGroupRunawayOption`, but now it will be reset to an empty new one if `len(opt.RunawayOptionList) > 0`. Will this be a problem?

Reply:
Nice catch.